lib/controllers/rocks.js

```diff
@@ -0,0 +1,26 @@
+const { Router } = require('express');
+const Rock = require('../models/Rock');
+module.exports = Router()
+.patch('/:id', async (req, res) => {
+  const rock = await Rock.update(req.params.id, req.body);
+  res.send(rock);
+} )
+.get('/:id', async (req, res, next) => {
+  try { 
+    const rock = await Rock.findById(req.params.id);
+  res.send(rock);
+  } catch (error) {
+    error.status = 404;
```

Comment:
We may want to make this error more specific, in the case of get by id, maybe only set the status to 404 if no item is returned from the findById function